src/CaptainHook.Application/Handlers/Subscribers/UpsertWebhookRequestHandler.cs

```
@@ -79,16 +86,18 @@ public UpsertWebhookRequestHandler(ISubscriberRepository subscriberRepository, I
             var endpoint = MapRequestToEndpoint(request, existingItem);
             existingItem.AddWebhookEndpoint(endpoint);
```

Validation does not take place inside entities. It gives us better control over the flow as we do not have to throw exceptions or inject validators into entities. It introduces incoherence however as entities can be invalid until proven otherwise. I am ok with this approach but let's ensure there is a buy-in from everyone (@chomick  :)?) before we invest more effort.

---

I think the entity should return an error if operation you perform on it is incorrect.
In this case, you should not be allowed to add new endpoint to the entity and the method should respond with OperationResult<> of some sort.

---

It's good I asked then. Let me see what I can do here. @tabman83 @dawidkeshop what do you reckon?

---

I am kinda split between the two, on one side I don't want those entities to be too fat or have the validation in two different places

---

Please take a look at the new version, validation of entities is no longer done on the handler level.